### PR TITLE
JAMES-2477 Allow passing additional arguments to maven dockerized build

### PR DESCRIPTION
**Pull request by chibenwa**
This should allow passing:

 - `-Dpartial-build` profile when needed
 - `-T 4` for parallel builds

### CODE REVIEW

File: dockerfiles/compilation/java-8/compile.sh [hunk 1]
```diff
@@ -47,9 +50,9 @@ git checkout $SHA1
 # Compilation
 
 if [ "$SKIPTESTS" = "skipTests" ]; then
-   mvn package -DskipTests
+   mvn package -DskipTests $MVN_ADDITIONAL_ARG_LINE
```
Comment (rouazana) [inline dockerfiles/compilation/java-8/compile.sh]: I fear you don't handle spaces in your parameters with this syntax. If I remember correctly you should write `"${MVN_ADDITIONAL_ARG_LINE}"` instead

Comment (chibenwa) [reply dockerfiles/compilation/java-8/compile.sh]: Nice to know!

File: dockerfiles/compilation/java-8/compile.sh [hunk 2]
```diff
@@ -50,9 +50,9 @@ git checkout $SHA1
 # Compilation
 
 if [ "$SKIPTESTS" = "skipTests" ]; then
-   mvn package -DskipTests $MVN_ADDITIONAL_ARG_LINE
+   mvn package -DskipTests ${MVN_ADDITIONAL_ARG_LINE}
```
Comment (rouazana) [inline dockerfiles/compilation/java-8/compile.sh]: you missed the double quotes around (yes, this syntax sux)

Comment (rouazana) [reply dockerfiles/compilation/java-8/compile.sh]: let's forget it for now, see:
```
       Only brace expansion, word splitting, and pathname expansion can change
       the number of words of the expansion; other expansions expand a  single
       word  to a single word.  The only exceptions to this are the expansions
       of "$@" and "${name[@]}" as explained above (see PARAMETERS).
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**